src/Widgets/calendar/CalendarModel.vala

```diff
@@ -121,20 +121,27 @@ namespace DateTime.Widgets {
         /* --- Public Methods ---// */
 
         public Gee.ArrayList<Event> get_events (GLib.DateTime date) {
-            var events_on_day = new Gee.TreeMap<string,Event> ();
-            foreach (var entry in source_events.get_values ()) {
-                foreach (var comp in entry.values) {
-                    unowned iCal.Component ical = comp.get_icalcomponent ();
-                    foreach (var dt_range in Util.event_date_ranges (ical, data_range)) {
-                        if (dt_range.contains (date)) {
-                            if (!events_on_day.has_key (ical.get_uid ())) {
-                                events_on_day.set (ical.get_uid (), new Event (date, dt_range, ical));
+            var events = new Gee.TreeMap<string,Event> ();
+            registry.list_sources (E.SOURCE_EXTENSION_CALENDAR).foreach ((source) => {
```

Comment:
So it kind of looks like we don't need to do this. In the code you deleted, there's a hashmap `source_events` that we're iterating through. So it kind of seems like we can get the source from there instead of having to list all sources again

Reply:
It's a different kind of Source. We want the sources with E.SOURCE_EXTENSION_CALENDAR enabled here. Can't get those from source_events, as those are just the events themselves.